Split segments

mem48 opened the pull request:
@Robinlovelace new PR that starts on splitting routes and segments.

The current version implements `segments = both` and then passes routes into the main workflow. Segments are saved for later.

As code currently expects there to be quietness and gradient variables in the routes, dummy data is provided until downstream code can be refactored

Robinlovelace commented:
Pre-requisite of merging this: fixed gradients and quietness.

Robinlovelace reviewed (changes requested):
@mem48 fails for me with

```
Uptake for fastest school routes
✖ error target uptake_list_school
▶ end pipeline [0.259 seconds]
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: no applicable method for 'mutate' applied to an object of class "list"
```

More importantly this would be a regression as it looses gradient and quietness. Let me try the 4x memory fix and we can revisit if that doesn't solve it is my current thinking on this.

Robinlovelace commented:
Will give it a spin, looks promising! Can you try #255 also if you get a chance. Happy with whichever works best.

Robinlovelace commented:
Build successful! Let me try on Windows laptop with large-ish build overnight...

Robinlovelace commented:
Heads-up @mem48 here are the results so far, seems to be taking *longer* than before on the rnet stages. I guess that's because adding more attributes to rnet as we did before has little overhead but now we're generating 2 sets of rnets for each plan. 

I predict the code in #255 will be a bit quicker but let's see.

<details>

```

> tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (6c0cccf0) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'odjitter' from a github remote, the SHA1 (a756bd4d) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'targets' from a cran remote, the SHA1 (1.2.2) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Linking to GEOS 3.11.2, GDAL 3.6.2, PROJ 9.2.0; sf_use_s2() is TRUE
▶ start target param_file
● built target param_file [0.48 seconds]
▶ start target parameters
● built target parameters [0.02 seconds]
▶ start target study_area
● built target study_area [0 seconds]
▶ start target r_school_rns
Getting the fastest routes for school journeys
POSTing the request to create and start the job
Job id: 5820
No data returned yet. Trying again id 5820 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5820: deleted
Saving outputdata/2023-08-15/routes_max_dist_school_fastest.Rds
Getting the balanced routes for school journeys
POSTing the request to create and start the job
Job id: 5821
No data returned yet. Trying again id 5821 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5821: deleted
Saving outputdata/2023-08-15/routes_max_dist_school_balanced.Rds
Getting the quietest routes for school journeys
POSTing the request to create and start the job
Job id: 5822
No data returned yet. Trying again id 5822 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5822: deleted
Saving outputdata/2023-08-15/routes_max_dist_school_quietest.Rds
Getting the ebike routes for school journeys
POSTing the request to create and start the job
Job id: 5823
No data returned yet. Trying again id 5823 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5823: deleted
Saving outputdata/2023-08-15/routes_max_dist_school_ebike.Rds
● built target r_school_rns [1.378 hours]
▶ start target zones
● built target zones [0.36 seconds]
▶ start target s_school
● built target s_school [0 seconds]
▶ start target r_school
● built target r_school [0 seconds]
▶ start target subpoints_origins
● built target subpoints_origins [0.06 seconds]
▶ start target od_data
● built target od_data [1.58 seconds]
▶ start target uptake_list_school
Uptake for fastest school routes
Uptake for balanced school routes
Uptake for quietest school routes
Uptake for ebike school routes
● built target uptake_list_school [2.458 minutes]
▶ start target subpoints_destinations
Getting destinations
● built target subpoints_destinations [1.2 seconds]
▶ start target rnet_school_list
Building Primary fastest network
2023-08-14 23:31:02.621723 constructing segments
2023-08-14 23:31:14.371547 building geometry
2023-08-14 23:31:26.167601 simplifying geometry
large data detected, using regionalisation, nrow = 754440
2023-08-14 23:32:00.772815 regionalisation complete, aggregating flows
2023-08-14 23:33:35.720576 rejoining segments into linestrings
Building Primary balanced network
2023-08-14 23:33:43.603509 constructing segments
2023-08-14 23:34:03.292531 building geometry
2023-08-14 23:34:14.095866 simplifying geometry
large data detected, using regionalisation, nrow = 767828
2023-08-14 23:34:48.694705 regionalisation complete, aggregating flows
2023-08-14 23:37:11.494043 rejoining segments into linestrings
Building Primary quietest network
2023-08-14 23:37:17.271651 constructing segments
2023-08-14 23:37:31.939503 building geometry
2023-08-14 23:37:44.255071 simplifying geometry
large data detected, using regionalisation, nrow = 778435
2023-08-14 23:38:26.324449 regionalisation complete, aggregating flows
2023-08-14 23:41:02.182291 rejoining segments into linestrings
Building Primary ebike network
2023-08-14 23:41:07.526188 constructing segments
2023-08-14 23:41:18.945713 building geometry
2023-08-14 23:41:29.706577 simplifying geometry
large data detected, using regionalisation, nrow = 752025
2023-08-14 23:42:05.527195 regionalisation complete, aggregating flows
2023-08-14 23:45:26.676562 rejoining segments into linestrings
Building Secondary fastest network
2023-08-14 23:45:32.285822 constructing segments
2023-08-14 23:45:44.45502 building geometry
2023-08-14 23:45:53.180391 simplifying geometry
large data detected, using regionalisation, nrow = 544097
2023-08-14 23:46:25.272756 regionalisation complete, aggregating flows
2023-08-14 23:49:21.378736 rejoining segments into linestrings
Building Secondary balanced network
2023-08-14 23:49:25.679001 constructing segments
2023-08-14 23:49:38.010552 building geometry
2023-08-14 23:49:45.737277 simplifying geometry
large data detected, using regionalisation, nrow = 540686
2023-08-14 23:50:10.427632 regionalisation complete, aggregating flows
2023-08-14 23:53:14.903341 rejoining segments into linestrings
Building Secondary quietest network
2023-08-14 23:53:20.697981 constructing segments
2023-08-14 23:53:32.798002 building geometry
2023-08-14 23:53:42.161694 simplifying geometry
large data detected, using regionalisation, nrow = 544954
2023-08-14 23:54:08.581145 regionalisation complete, aggregating flows
2023-08-14 23:57:03.258881 rejoining segments into linestrings
Building Secondary ebike network
2023-08-14 23:57:08.986234 constructing segments
2023-08-14 23:57:21.3583 building geometry
2023-08-14 23:57:37.200867 simplifying geometry
large data detected, using regionalisation, nrow = 541988
2023-08-14 23:58:01.565766 regionalisation complete, aggregating flows
2023-08-15 00:00:45.836765 rejoining segments into linestrings
● built target rnet_school_list [29.91 minutes]
▶ start target od_commute_jittered
Scraped 6976 zones from C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/zones.geojson
Scraped 46351 subpoints from C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/subpoints_origins.geojson
Scraped 179210 subpoints from C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/subpoints_destinations.geojson
Disaggregating OD data
Wrote C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/od_jittered.geojson
● built target od_commute_jittered [3.114 minutes]
▶ start target od_commute_subset
● built target od_commute_subset [0.76 seconds]
▶ start target r_commute_rns
2023-08-15
Calculating 100000 routes
Getting the fastest routes for commute journeys
POSTing the request to create and start the job
Job id: 5824
No data returned yet. Trying again id 5824 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916
When using segments = both, only the routes are joined to desire lines
28 routes removed
Job 5824: deleted
Saving outputdata/2023-08-15/routes_max_dist_commute_fastest.Rds
Getting the balanced routes for commute journeys
POSTing the request to create and start the job
Job id: 5825
No data returned yet. Trying again id 5825 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916
When using segments = both, only the routes are joined to desire lines
28 routes removed
Job 5825: deleted
Saving outputdata/2023-08-15/routes_max_dist_commute_balanced.Rds
Getting the quietest routes for commute journeys
POSTing the request to create and start the job
Job id: 5826
No data returned yet. Trying again id 5826 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916
When using segments = both, only the routes are joined to desire lines
28 routes removed
Job 5826: deleted
Saving outputdata/2023-08-15/routes_max_dist_commute_quietest.Rds
Getting the ebike routes for commute journeys
POSTing the request to create and start the job
Job id: 5827
No data returned yet. Trying again id 5827 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\georl\AppData\Local\Temp\Rtmp80IlpA/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916
When using segments = both, only the routes are joined to desire lines
28 routes removed
Job 5827: deleted
Saving outputdata/2023-08-15/routes_max_dist_commute_ebike.Rds
● built target r_commute_rns [2.787 hours]
▶ start target r_commute
● built target r_commute [0 seconds]
▶ start target s_commute
● built target s_commute [0 seconds]
▶ start target uptake_list_commute
Uptake for fastest
Uptake for balanced
Uptake for quietest
Uptake for ebike
● built target uptake_list_commute [5.882 minutes]
▶ start target calculate_benefits
● built target calculate_benefits [0 seconds]
▶ start target rnet_segments
2023-08-15 03:09:27.212381 constructing segments
2023-08-15 03:28:20.378273 building geometry
2023-08-15 03:29:44.399557 simplifying geometry
2023-08-15 03:29:44.407854 aggregating flows
2023-08-15 03:32:11.259017 rejoining segments into linestrings
● built target rnet_segments [28.88 minutes]
▶ start target zones_stats_list
`summarise()` has grouped output by 'route_id'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'geo_code1'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'geo_code2'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'route_id', 'plan'. You can override using
the `.groups` argument.
`summarise()` has grouped output by 'DataZone', 'plan'. You can override using
the `.groups` argument.
`summarise()` has grouped output by 'SeedCode', 'plan'. You can override using
the `.groups` argument.
● built target zones_stats_list [50.67 seconds]
▶ start target rnet_commute_list
Building Commute fastest network
2023-08-15 03:33:47.928596 constructing segments
2023-08-15 03:36:39.499869 building geometry
2023-08-15 03:37:19.720943 simplifying geometry
large data detected, using regionalisation, nrow = 1279552
2023-08-15 03:39:24.980609 regionalisation complete, aggregating flows
2023-08-15 04:01:16.239357 rejoining segments into linestrings
Building Commute balanced network
2023-08-15 04:01:47.524424 constructing segments
2023-08-15 04:05:32.297543 building geometry
2023-08-15 04:06:28.637008 simplifying geometry
large data detected, using regionalisation, nrow = 1309918
2023-08-15 04:07:49.617805 regionalisation complete, aggregating flows
2023-08-15 04:34:12.158083 rejoining segments into linestrings
Building Commute quietest network
2023-08-15 04:34:42.858443 constructing segments
2023-08-15 04:38:49.775593 building geometry
2023-08-15 04:40:00.661145 simplifying geometry
large data detected, using regionalisation, nrow = 1331528
2023-08-15 04:41:16.825012 regionalisation complete, aggregating flows
2023-08-15 05:08:35.971095 rejoining segments into linestrings
Building Commute ebike network
2023-08-15 05:09:06.935034 constructing segments
2023-08-15 05:12:34.565751 building geometry
2023-08-15 05:13:12.849364 simplifying geometry
large data detected, using regionalisation, nrow = 1274995
2023-08-15 05:14:23.835378 regionalisation complete, aggregating flows
2023-08-15 05:56:23.713833 rejoining segments into linestrings
● built target rnet_commute_list [2.388 hours]
▶ start target zones_stats
● built target zones_stats [0 seconds]
▶ start target school_stats
● built target school_stats [0 seconds]
▶ start target combined_network
2023-08-15 05:59:32.987183 constructing segments
2023-08-15 06:18:38.173306 building geometry
2023-08-15 06:19:31.34229 simplifying geometry
large data detected, using regionalisation, nrow = 1802185
2023-08-15 06:22:13.528412 regionalisation complete, aggregating flows
```

It's still running after running all night and we're only on max = 100k routes.

Robinlovelace commented:
Also peak memory use above 90% on my computer.

Robinlovelace commented on _targets.R:
```diff
@@ -330,11 +366,14 @@ list(
-    rnet_combined = combine_rnets(rnl = c(rnet_cl, rnet_sl_p, rnet_sl_s),
+    rnl = c(rnet_cl, rnet_sl_p, rnet_sl_s, list(rnet_quietness))
+    
+    rnet_combined = combine_rnets(rnl = rnl,
```
This stage seems to be taking forever, more than an hour, on my laptop:

```
▶ start target combined_network
2023-08-15 05:59:32.987183 constructing segments
2023-08-15 06:18:38.173306 building geometry
2023-08-15 06:19:31.34229 simplifying geometry
large data detected, using regionalisation, nrow = 1802185
2023-08-15 06:22:13.528412 regionalisation complete, aggregating flows
```

mem48 replied:
I only see 24 minutes in the times? Whats the progress on the last stage?

Robinlovelace replied:
It had hanged for ~60 minutes by the time I posted that message. Key question: can you reproduce this behaviour?

mem48 replied:
Can you share the routes? Would be good to have a large scale test dataset on teams.

Robinlovelace replied:
> Can you share the routes? Would be good to have a large scale test dataset on teams.

Sure but not on the home laptop ATM. Just run `tar_make()` and wait a few minutes on this branch.

Robinlovelace reviewed (commented):
Doesn't seem to have led to a substantial speed-up or reduction in memory use. Can you see if you can reproduce this performance (8+ hours with only 100k routes), to eliminate the possibility that it's just my computer @mem48 ? In the mean time we can also benchmark #255 .

mem48 commented:
This seems very odd. Can you post summaries of the inputs to the rnet merge.

mem48 commented:
Also are you getting the progress bar with the time estimates?

mem48 commented:
Have you seen https://books.ropensci.org/targets/performance.html#memory

Robinlovelace commented:
> Also are you getting the progress bar with the time estimates?

See details above and reproduce this behaviour by pulling the latest changes on this branch, checking `outputdata/2023-08-15` is empty and running

```r
targets::tar_make()
```

Robinlovelace commented:
> Have you seen https://books.ropensci.org/targets/performance.html#memory

No, this is worth doing:

```r
tar_option_set(memory = "transient", garbage_collection = TRUE)
```

Robinlovelace commented:
```
● built target rnet_commute_list [2.388 hours]
```

In the above suggests that after this PR route network generation takes the same amount of time as routing.

Robinlovelace commented:
And that's before the combined network stage..

mem48 commented:
Thinking about this if you are hitting 90%+ memory usage your probably using the page file, which massively reduces performance.

Could also look at replacing dplyr with data.table in overline

Robinlovelace commented:
> Thinking about this if you are hitting 90%+ memory usage your probably using the page file, which massively reduces performance.
> 
> Could also look at replacing dplyr with data.table in overline

Sounds good but not a priority. Let's see how #255 gets on, currently running on same laptop.

mem48 commented:
Full build success


```
> targets::tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (6c0cccf0) has not changed since last install.
  Use `force = TRUE` to force installation
Downloading GitHub repo dabreegster/odjitter@HEAD
|nits (0.8-2 -> 0.8-3) [CRAN]
Installing 1 packages: units
trying URL 'https://cran.rstudio.com/bin/windows/contrib/4.2/units_0.8-3.zip'
 length 795995 bytes (777 KB)'
==========
downloaded 777 KB

package 'units' successfully unpacked and MD5 sums checked
-
The downloaded binary packages are in
	C:\Users\earmmor\AppData\Local\Temp\RtmpEF4J5R\downloaded_packages
-─ R CMD build ─────────────────────────────────────────────────────────────────
/  checking for file 'C:\Users\earmmor\AppData\Local\Temp\RtmpEF4J5R\remotes5a18e1e4911\dabreegster-odjitter-2659dd2\r/DESCRIPTION' (646ms)
/  preparing 'odjitter':
-  checking DESCRIPTION meta-information ... 
-  checking for LF line-endings in source and make files and shell scripts
|  checking for empty or unneeded directories
|  building 'odjitter_0.0.0.9000.tar.gz'
   
* installing *source* package 'odjitter' ...
** using staged installation
** R
** byte-compile and prepare package for lazy loading
** help
*** installing help indices
** building package indices
** testing if installed package can be loaded from temporary location
** testing if installed package can be loaded from final location
** testing if installed package keeps a record of temporary installation path
* DONE (odjitter)
Skipping install of 'stplanr' from a github remote, the SHA1 (6583e631) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'targets' from a cran remote, the SHA1 (1.2.2) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
▶ start target param_file
● built target param_file [1.34 seconds]
▶ start target parameters
● built target parameters [0 seconds]
▶ start target study_area
● built target study_area [0 seconds]
▶ start target r_school_rns
Getting the fastest routes for school journeys
POSTing the request to create and start the job
Job id: 5846
\
> targets::tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (6c0cccf0) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'odjitter' from a github remote, the SHA1 (2659dd21) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'stplanr' from a github remote, the SHA1 (6583e631) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'targets' from a cran remote, the SHA1 (1.2.2) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
✔ skip target param_file
✔ skip target parameters
✔ skip target study_area
▶ start target r_school_rns
Getting the fastest routes for school journeys
POSTing the request to create and start the job
Job id: 5847
No data returned yet. Trying again id 5847 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5847: deleted
Saving outputdata/2023-08-17/routes_max_dist_school_fastest.Rds
Getting the balanced routes for school journeys
POSTing the request to create and start the job
Job id: 5848
No data returned yet. Trying again id 5848 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5848: deleted
Saving outputdata/2023-08-17/routes_max_dist_school_balanced.Rds
Getting the quietest routes for school journeys
POSTing the request to create and start the job
Job id: 5849
No data returned yet. Trying again id 5849 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5849: deleted
Saving outputdata/2023-08-17/routes_max_dist_school_quietest.Rds
Getting the ebike routes for school journeys
POSTing the request to create and start the job
Job id: 5850
No data returned yet. Trying again id 5850 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/test.csv.gz
When using segments = both, only the routes are joined to desire lines
0 routes removed
Job 5850: deleted
Saving outputdata/2023-08-17/routes_max_dist_school_ebike.Rds
● built target r_school_rns [1.472 hours]
▶ start target zones
● built target zones [0.47 seconds]
▶ start target r_school
● built target r_school [0 seconds]
▶ start target s_school
● built target s_school [0 seconds]
▶ start target subpoints_origins
● built target subpoints_origins [0.19 seconds]
▶ start target od_data
● built target od_data [2 seconds]
▶ start target uptake_list_school
Uptake for fastest school routes
Uptake for balanced school routes
Uptake for quietest school routes
Uptake for ebike school routes
● built target uptake_list_school [3.889 minutes]
▶ start target subpoints_destinations
Getting destinations
● built target subpoints_destinations [1.52 seconds]
▶ start target rnet_school_list
Building Primary fastest network
2023-08-17 11:34:44 constructing segments
2023-08-17 11:35:03 building geometry
2023-08-17 11:35:20 simplifying geometry
large data detected, using regionalisation, nrow = 754576
2023-08-17 11:36:14 regionalisation complete, aggregating flows
2023-08-17 11:36:46 rejoining segments into linestrings
Building Primary balanced network
2023-08-17 11:36:54 constructing segments
2023-08-17 11:37:16 building geometry
2023-08-17 11:37:32 simplifying geometry
large data detected, using regionalisation, nrow = 767696
2023-08-17 11:38:25 regionalisation complete, aggregating flows
2023-08-17 11:39:00 rejoining segments into linestrings
Building Primary quietest network
2023-08-17 11:39:07 constructing segments
2023-08-17 11:39:30 building geometry
2023-08-17 11:39:46 simplifying geometry
large data detected, using regionalisation, nrow = 778242
2023-08-17 11:40:44 regionalisation complete, aggregating flows
2023-08-17 11:41:18 rejoining segments into linestrings
Building Primary ebike network
2023-08-17 11:41:25 constructing segments
2023-08-17 11:41:48 building geometry
2023-08-17 11:42:05 simplifying geometry
large data detected, using regionalisation, nrow = 752019
2023-08-17 11:43:04 regionalisation complete, aggregating flows
2023-08-17 11:43:32 rejoining segments into linestrings
Building Secondary fastest network
2023-08-17 11:43:41 constructing segments
2023-08-17 11:43:57 building geometry
2023-08-17 11:44:09 simplifying geometry
large data detected, using regionalisation, nrow = 544424
2023-08-17 11:44:47 regionalisation complete, aggregating flows
2023-08-17 11:45:09 rejoining segments into linestrings
Building Secondary balanced network
2023-08-17 11:45:15 constructing segments
2023-08-17 11:45:34 building geometry
2023-08-17 11:45:46 simplifying geometry
large data detected, using regionalisation, nrow = 540727
2023-08-17 11:46:24 regionalisation complete, aggregating flows
2023-08-17 11:46:46 rejoining segments into linestrings
Building Secondary quietest network
2023-08-17 11:46:51 constructing segments
2023-08-17 11:47:14 building geometry
2023-08-17 11:47:26 simplifying geometry
large data detected, using regionalisation, nrow = 544927
2023-08-17 11:48:04 regionalisation complete, aggregating flows
2023-08-17 11:48:27 rejoining segments into linestrings
Building Secondary ebike network
2023-08-17 11:48:35 constructing segments
2023-08-17 11:48:52 building geometry
2023-08-17 11:49:05 simplifying geometry
large data detected, using regionalisation, nrow = 542016
2023-08-17 11:49:44 regionalisation complete, aggregating flows
2023-08-17 11:50:10 rejoining segments into linestrings
● built target rnet_school_list [15.65 minutes]
▶ start target od_commute_jittered
Scraped 6976 zones from C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/zones.geojson
Scraped 46351 subpoints from C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/subpoints_origins.geojson
Scraped 179210 subpoints from C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/subpoints_destinations.geojson
Disaggregating OD data
Wrote C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/od_jittered.geojson
● built target od_commute_jittered [2.152 minutes]
▶ start target od_commute_subset
● built target od_commute_subset [1.39 seconds]
▶ start target r_commute_rns
2023-08-17
Calculating 433150 routes
Getting the fastest routes for commute journeys
POSTing the request to create and start the job
Job id: 5851
No data returned yet. Trying again id 5851 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\Rtmp422ovE/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916 113738 115608 122278 122295 126869 133017 139848 139852 140005 140221 145732 164441 165937 186149 189960 202512 211857 214316 214788 214803 214922 215838 216312 216727 231494 231687 240185 240235 240294 242021 242657 262992 305233 306175 353283 353421 353621 353951 363022 384401 384487 384621 386054 386229 396469 396971 397651 398001 398934 402097 420322 427888 428514 428877 430169
When using segments = both, only the routes are joined to desire lines
83 routes removed
Job 5851: deleted
Saving outputdata/2023-08-17/routes_max_dist_commute_fastest.Rds
Getting the balanced routes for commute journeys
POSTing the request to create and start the job
Job id: 5853
No data returned yet. Trying again id 5853 every 10 seconds
Sending data, wait...
Routing not complete
✖ error target r_commute_rns
▶ end pipeline [5.932 hours]
|here were 11 warnings (use warnings() to see them)
Error:
! Error running targets::tar_make()
  Error messages: targets::tar_meta(fields = error, complete_only = TRUE)
  Debugging guide: https://books.ropensci.org/targets/debugging.html
  How to ask for help: https://books.ropensci.org/targets/help.html
  Last error: Timeout was reached: [api.cyclestreets.net] Send failure: Connection was reset
> usethis::edit_r_environ()
• Modify 'C:/Users/earmmor/.Renviron'
• Restart R for changes to take effect
> ?cyclestreets::batch
> Sys.getenv("CYCLESTREETS_PW")
[1] "cruse2022"
> tar_make()
Error in tar_make() : could not find function "tar_make"
> library(targets)
> tar_make()
Skipping install of 'cyclestreets' from a github remote, the SHA1 (6c0cccf0) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'odjitter' from a github remote, the SHA1 (2659dd21) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'stplanr' from a github remote, the SHA1 (6583e631) has not changed since last install.
  Use `force = TRUE` to force installation
Skipping install of 'targets' from a cran remote, the SHA1 (1.2.2) has not changed since last install.
  Use `force = TRUE` to force installation
── Attaching core tidyverse packages ──────────────────────── tidyverse 2.0.0 ──
✔ dplyr     1.1.2     ✔ readr     2.1.4
✔ forcats   1.0.0     ✔ stringr   1.5.0
✔ ggplot2   3.4.2     ✔ tibble    3.2.1
✔ lubridate 1.9.2     ✔ tidyr     1.3.0
✔ purrr     1.0.1     
── Conflicts ────────────────────────────────────────── tidyverse_conflicts() ──
✖ dplyr::filter() masks stats::filter()
✖ dplyr::lag()    masks stats::lag()
ℹ Use the conflicted package (<http://conflicted.r-lib.org/>) to force all conflicts to become errors
Linking to GEOS 3.9.3, GDAL 3.5.2, PROJ 8.2.1; sf_use_s2() is TRUE
✔ skip target param_file
✔ skip target parameters
✔ skip target study_area
✔ skip target r_school_rns
✔ skip target zones
✔ skip target r_school
✔ skip target s_school
✔ skip target subpoints_origins
✔ skip target od_data
✔ skip target uptake_list_school
✔ skip target subpoints_destinations
✔ skip target rnet_school_list
✔ skip target od_commute_jittered
✔ skip target od_commute_subset
▶ start target r_commute_rns
2023-08-17
Calculating 433150 routes
Getting the fastest routes for commute journeys
Routes already exist: reading from file: outputdata/2023-08-17/routes_max_dist_commute_fastest.Rds
Getting the balanced routes for commute journeys
POSTing the request to create and start the job
Job id: 5854
No data returned yet. Trying again id 5854 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpWyedUl/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916 113738 115608 122278 122295 126869 133017 139848 139852 140005 140221 145732 164441 165937 186149 189960 202512 211857 214316 214788 214803 214922 215838 216312 216727 231494 231687 240185 240235 240294 242021 242657 262992 305233 306175 353283 353421 353621 353951 363022 384401 384487 384621 386054 386229 396469 396971 397651 398001 398934 402097 420322 427888 428514 428877 430169
When using segments = both, only the routes are joined to desire lines
83 routes removed
Job 5854: deleted
Saving outputdata/2023-08-17/routes_max_dist_commute_balanced.Rds
Getting the quietest routes for commute journeys
POSTing the request to create and start the job
Job id: 5855
No data returned yet. Trying again id 5855 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpWyedUl/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916 113738 115608 122278 122295 126869 133017 139848 139852 140005 140221 145732 164441 165937 186149 189960 202512 211857 214316 214788 214803 214922 215838 216312 216727 231494 231687 240185 240235 240294 242021 242657 262992 305233 306175 353283 353421 353621 353951 363022 384401 384487 384621 386054 386229 396469 396971 397651 398001 398934 402097 420322 427888 428514 428877 430169
When using segments = both, only the routes are joined to desire lines
83 routes removed
Job 5855: deleted
Saving outputdata/2023-08-17/routes_max_dist_commute_quietest.Rds
Getting the ebike routes for commute journeys
POSTing the request to create and start the job
Job id: 5856
No data returned yet. Trying again id 5856 every 10 seconds
Sending data, wait...
Routing not complete
Congrats, your data is ready
test already exists, overwriting it
Reading in the following file:
C:\Users\earmmor\AppData\Local\Temp\RtmpWyedUl/test.csv.gz
Removing NA routes: 4931 7931 8855 12542 14292 25738 28302 29277 32959 35373 38796 41561 47187 48501 59438 68319 68407 73334 79324 79366 82467 84317 86292 86664 86804 93451 93468 95916 113738 115608 122278 122295 126869 133017 139848 139852 140005 140221 145732 164441 165937 186149 189960 202512 211857 214316 214788 214803 214922 215838 216312 216727 231494 231687 240185 240235 240294 242021 242657 262992 305233 306175 353283 353421 353621 353951 363022 384401 384487 384621 386054 386229 396469 396971 397651 398001 398934 402097 420322 427888 428514 428877 430169
When using segments = both, only the routes are joined to desire lines
83 routes removed
Job 5856: deleted
Saving outputdata/2023-08-17/routes_max_dist_commute_ebike.Rds
● built target r_commute_rns [11.119 hours]
▶ start target r_commute
● built target r_commute [0 seconds]
▶ start target s_commute
● built target s_commute [0 seconds]
▶ start target uptake_list_commute
Uptake for fastest
Uptake for balanced
Uptake for quietest
Uptake for ebike
● built target uptake_list_commute [31.438 minutes]
▶ start target calculate_benefits
● built target calculate_benefits [0 seconds]
▶ start target rnet_segments
2023-08-18 05:48:39 constructing segments
2023-08-18 07:55:42 building geometry
2023-08-18 08:01:41 simplifying geometry
2023-08-18 08:01:41 aggregating flows
2023-08-18 08:10:01 rejoining segments into linestrings
● built target rnet_segments [2.819 hours]
▶ start target zones_stats_list
`summarise()` has grouped output by 'route_id'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'geo_code1'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'geo_code2'. You can override using the
`.groups` argument.
`summarise()` has grouped output by 'route_id', 'plan'. You can override using
the `.groups` argument.
`summarise()` has grouped output by 'DataZone', 'plan'. You can override using
the `.groups` argument.
`summarise()` has grouped output by 'SeedCode', 'plan'. You can override using
the `.groups` argument.
● built target zones_stats_list [1.673 minutes]
▶ start target rnet_commute_list
Building Commute fastest network
2023-08-18 09:05:17 constructing segments
2023-08-18 09:12:06 building geometry
2023-08-18 09:12:36 simplifying geometry
large data detected, using regionalisation, nrow = 1636097
2023-08-18 09:14:59 regionalisation complete, aggregating flows
2023-08-18 09:19:04 rejoining segments into linestrings
Building Commute balanced network
2023-08-18 09:20:10 constructing segments
2023-08-18 09:25:52 building geometry
2023-08-18 09:26:24 simplifying geometry
large data detected, using regionalisation, nrow = 1690691
2023-08-18 09:31:13 regionalisation complete, aggregating flows
2023-08-18 09:33:16 rejoining segments into linestrings
Building Commute quietest network
2023-08-18 09:34:38 constructing segments
2023-08-18 09:43:37 building geometry
2023-08-18 09:44:08 simplifying geometry
large data detected, using regionalisation, nrow = 1722517
2023-08-18 09:46:18 regionalisation complete, aggregating flows
2023-08-18 09:53:17 rejoining segments into linestrings
Building Commute ebike network
2023-08-18 09:54:27 constructing segments
2023-08-18 09:59:39 building geometry
2023-08-18 10:00:08 simplifying geometry
large data detected, using regionalisation, nrow = 1616355
2023-08-18 10:02:09 regionalisation complete, aggregating flows
2023-08-18 10:06:25 rejoining segments into linestrings
● built target rnet_commute_list [1.037 hours]
▶ start target school_stats
● built target school_stats [0 seconds]
▶ start target zones_stats
● built target zones_stats [0.02 seconds]
▶ start target combined_network
2023-08-18 10:10:20 constructing segments
2023-08-18 10:17:56 building geometry
2023-08-18 10:18:39 simplifying geometry
large data detected, using regionalisation, nrow = 2101669
2023-08-18 10:27:42 regionalisation complete, aggregating flows
2023-08-18 10:34:16 rejoining segments into linestrings
● built target combined_network [26.228 minutes]
▶ start target school_stats_json
Writing JSON
Zipping JSON
● built target school_stats_json [58.89 seconds]
▶ start target zones_stats_json
Writing JSON
Zipping JSON
● built target zones_stats_json [2.482 minutes]
▶ start target combined_network_tile
● built target combined_network_tile [0 seconds]
▶ start target save_outputs
Saving outputs for 2023-08-17
outputdata/combined_network_tile.geojson' failed
combined_network_tile' to data source 
  `outputdata/combined_network_tile.geojson' using driver `GeoJSON'
546406 features with 26 fields and geometry type Line String.
● built target save_outputs [1.656 minutes]
▶ start target upload_data
Commit: cbb84b024550d638dbca066c5850d1b03d55fc66Malcolm Morgan <mem48@users.noreply.github.com>16920401052FALSEWorking without dummy data

Creating new release and folder to save the files: v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66
Going to try to upload the following files: combined_network_tile.zip, DataZone_json.zip, od_commute_subset.Rds, rnet_commute_list.Rds, rnet_primary_school_balanced.Rds, rnet_primary_school_ebike.Rds, rnet_primary_school_fastest.Rds, rnet_primary_school_quietest.Rds, rnet_school_list.Rds, rnet_secondary_school_balanced.Rds, rnet_secondary_school_ebike.Rds, rnet_secondary_school_fastest.Rds, rnet_secondary_school_quietest.Rds, routes_commute_balanced.Rds, routes_commute_ebike.Rds, routes_commute_fastest.Rds, routes_commute_quietest.Rds, school_stats.Rds, SeedCode_json.zip, zones_stats.Rds
With sizes:  25.54M,  10.09M,  19.36M,  72.95M,   7.29M,   7.21M,   7.26M,   7.33M,  49.75M,   5.12M,   5.18M,   5.22M,   5.12M, 658.1M, 604.37M, 589.45M, 679.63M, 676.83K,   1.35M,  45.77M
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 combined_network_tile.zip --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 DataZone_json.zip --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 od_commute_subset.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_commute_list.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_primary_school_balanced.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_primary_school_ebike.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_primary_school_fastest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_primary_school_quietest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_school_list.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_secondary_school_balanced.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_secondary_school_ebike.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_secondary_school_fastest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 rnet_secondary_school_quietest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 routes_commute_balanced.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 routes_commute_ebike.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 routes_commute_fastest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 routes_commute_quietest.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 school_stats.Rds --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 SeedCode_json.zip --clobber
Running this command:
gh release upload v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 zones_stats.Rds --clobber
Files stored in output folder: v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66
Which contains: combined_network_tile.zip, DataZone_json.zip, od_commute_subset.Rds, rnet_commute_list.Rds, rnet_primary_school_balanced.Rds, rnet_primary_school_ebike.Rds, rnet_primary_school_fastest.Rds, rnet_primary_school_quietest.Rds, rnet_school_list.Rds, rnet_secondary_school_balanced.Rds, rnet_secondary_school_ebike.Rds, rnet_secondary_school_fastest.Rds, rnet_secondary_school_quietest.Rds, routes_commute_balanced.Rds, routes_commute_ebike.Rds, routes_commute_fastest.Rds, routes_commute_quietest.Rds, school_stats.Rds, SeedCode_json.zip, zones_stats.Rds
● built target upload_data [1.148 minutes]
▶ start target metadata
Rows: 36 Columns: 6
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
dbl  (5): n_segment_cells, min_flow, max_to_route, time_total_mins, time_r_c...
date (1): routing_date
\
ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.
● built target metadata [1.8 seconds]
▶ end pipeline [18.224 hours]
|here were 35 warnings (use warnings() to see them)
```

mem48 commented:
@Robinlovelace new build on website, please check and approve this PR

Robinlovelace commented:
Great job!

Robinlovelace commented:
This contains a regression: low flows are plotted after high flows.

Before: 

![image](https://github.com/nptscot/npt/assets/1825120/570b3e11-ec9f-46e7-9c29-871fd9078619)

Source: https://www.npt.scot/#14.25/55.96425/-3.23035


After: 

![image](https://github.com/nptscot/npt/assets/1825120/4ab9c521-68c5-46c4-a50d-91e917a65986)

Source: https://nptscot.github.io/#14.25/55.96425/-3.23035

Robinlovelace commented:
Another example: 
![image](https://github.com/nptscot/npt/assets/1825120/e4630fcf-092f-4c3f-883a-677475ed858a)

Robinlovelace commented:
Will try to compare with results in here also: https://github.com/nptscot/outputdata/releases/tag/v2023-08-16-15-24-49.064012_commit_d23498806c464bb51c9e0a28129ca499010ececd

from #255

Robinlovelace commented:
Can you try to release the data, e.g. to https://github.com/nptscot/outputdata/releases/tag/v2023-08-18-10-42-44_commit_cbb84b024550d638dbca066c5850d1b03d55fc66 ?

Robinlovelace commented:
Just building the network from #255 with...


```

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

unzip("combined_network_tile.zip", exdir = "outputdata")
> unzip("combined_network_tile.zip", exdir = "outputdata")
> msg_verbose = paste0(
+       "--name=rnet --layer=rnet --attribution=UniverstyofLeeds --minimum-zoom=6 ",
+       "--maximum-zoom=13 --drop-smallest-as-needed --maximum-tile-bytes=5000000 ",
+       "--simplification=10 --buffer=5 --force  outputdata/combined_network_tile.geojson"
+     )
> targets::tar_load(parameters)
> date_routing = parameters$date_routing
> msg = glue::glue("tippecanoe -o outputdata/rnet_{date_routing}.pmtiles")
> system(paste(msg, msg_verbose))
outputdata/combined_network_tile.geojson: No such file or directory
0 features, 4 bytes of geometry, 0 bytes of string pool
Did not read any valid geometries
> !file.exists("outputdata/combined_network_tile.geojson")
[1] TRUE
> # Tile the data:
>     # system("bash code/tile.sh")
>     # # Manually get geojson:
>     # cd outputdata
>     # gh release download z2023-07-28 --pattern *.geojson
>     # cd ..
>     unzip("combined_network_tile.zip", exdir = ".")
> msg_verbose = paste0(
+       "--name=rnet --layer=rnet --attribution=UniverstyofLeeds --minimum-zoom=6 ",
+       "--maximum-zoom=13 --drop-smallest-as-needed --maximum-tile-bytes=5000000 ",
+       "--simplification=10 --buffer=5 --force  outputdata/combined_network_tile.geojson"
+     )
> date_routing = parameters$date_routing
> msg = glue::glue("tippecanoe -o outputdata/rnet_{date_routing}.pmtiles")
> system(paste(msg, msg_verbose))
522218 features, 67581427 bytes of geometry, 20691 bytes of string pool
tile 6/31/19 has 263151 features, >200000    
Going to try keeping the biggest 57.00% of the features to make it fit
tile 6/31/19 has 2
```

Robinlovelace commented:
Results from #255 

![image](https://github.com/nptscot/npt/assets/1825120/5624225d-9cc0-4ee0-be6c-b2e1edb4a355)

Results currently on https://nptscot.github.io/#14.25/55.96425/-3.23035

![image](https://github.com/nptscot/npt/assets/1825120/1760aee2-b0a0-4070-8e24-f92a09b3ff95)

Robinlovelace commented:
From that: it seems the results are identical. So no issue there.

Robinlovelace reviewed (changes requested):
Looks good from screenshots above. Only changes requested:

- [x] Release the data to a release in outputdata as mentioned above
- [x] Commit the code with the results in `build_summary.csv`

Then we're good to go!

Robinlovelace commented:
I think these files need to be updated:

https://github.com/nptscot/npt/blob/split-segments/parameters.json

https://github.com/nptscot/npt/blob/split-segments/outputs/build_summary.csv

https://github.com/nptscot/npt/blob/split-segments/outputs/metadata_targets.csv

Robinlovelace commented on outputs/metadata_targets.csv:
```diff
@@ -1,32 +1,41 @@
+zones_stats_json,stem,d8342b208032c13b,e77034c82e3c15ce,5840fdd9407d6a23,2033818370,,2023-08-18T09:41:50Z,887021b347b02e8f,100,rds,local,vector,NA,,148.93,NA,NA
+combined_network_tile,stem,8fc5c3c1a927d4b7,bcc0893adf783794,498896194ca3102c,-856067121,,2023-08-18T09:42:09Z,263df933483b3895,34632640,rds,local,vector,NA,,0,NA,NA
+save_outputs,stem,3d2538e3e694c41e,88a5e3c967bd1eb5,bb05f611eeeaf818,1635958269,,2023-08-18T09:44:04Z,dfd0d40dc0c5171b,95,rds,local,vector,NA,,99.35,"Thats a lot of columns for the GeoJSON, are they all needed. tippecanoe not found",NA
+upload_data,stem,ad58324e82f0d594,6310d8a851a9c1c6,b93d163c50cd3ff6,412588227,,2023-08-18T09:46:08Z,e655bbbdcbba1d63,81,rds,local,vector,NA,,68.88,gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found. gh not found.  ...
```
The `gh not found.` bit here explains why there was not auto release.

Robinlovelace commented on tippecanoe:
```diff
@@ -16,7 +16,7 @@ tippecanoe -o rnet-dev.pmtiles \
   --simplification=10 \
   --buffer=5 \
-  --force  rnet_school_commute.geojson
+  --force  combined_network_tile.geojson
```
Do we need to change the rnet tile code in `_targets.R`?

mem48 replied:
No

Robinlovelace reviewed (changes requested):
Something not right in `parameters.json`

Robinlovelace commented:
https://github.com/nptscot/npt/blob/d3419f813ef4774c4046469b7b95c565d897548c/parameters.json

mem48 commented:
Fixed

Robinlovelace commented on parameters.json:
```diff
@@ -1,13 +1,10 @@
   "plans": ["fastest", "balanced", "quietest", "ebike"],
   "min_flow": [1],
-<<<<<<< Updated upstream
   "max_to_route": [100000],
```
Shouldn't this be more like 10000000 ?

Robinlovelace commented:
Going to take the initiative and merge this. Great work Malcolm!
